src/axom/spin/MortonIndex.hpp

```diff
@@ -126,10 +126,14 @@ template <typename CoordType, typename MortonIndexType, typename Derived>
 struct MortonBase
 {
   // static assert to ensure we only instantiate on integral types
-  AXOM_STATIC_ASSERT_MSG(std::is_integral<CoordType>::value,
-                         "Coordtype must be integral for Morton indexing");
-  AXOM_STATIC_ASSERT_MSG(std::is_integral<MortonIndexType>::value,
-                         "MortonIndexType must be integral for Morton indexing");
+  static_assert(std::is_integral<CoordType>::value,
+                "Coordtype must be integral for Morton indexing");
+  static_assert(std::is_integral<MortonIndexType>::value,
+                "MortonIndexType must be integral for Morton indexing");
+
+  // we get undefined behavior w/ our bit shifts if using signed types
+  static_assert(std::is_unsigned<MortonIndexType>::value,
+                "MortonIndexType must be an unsigned type");
```

Comment:
This static_assert ensures that the `MortonIndexType` is unsigned. The only case within axom where it was a signed type was the BVH construction, which was triggering undefined behavior in a user code.